### PR TITLE
Fix iterator#next to return this

### PR DESCRIPTION
**Pull request by vweevers**
Fixes https://github.com/Level/leveldown/issues/443#issuecomment-364437129. I also updated the TypeScript typings @Level/typescript.

Should be released asap.

### CODE REVIEW

**Comment by vweevers** [comment]
I'm testing it against leveldown.

File: abstract-iterator.js [hunk 1]
```diff
@@ -14,18 +14,22 @@ AbstractIterator.prototype.next = function (callback) {
...
+    return self
   }
 
   self._nexting = true
   self._next(function () {
     self._nexting = false
     callback.apply(null, arguments)
   })
+
+  return self
```
Comment (ralphtheninja) [inline abstract-iterator.js]: Aah it was chaining that was the problem.

File: abstract/iterator-test.js [hunk 2]
```diff
@@ -40,6 +40,13 @@ module.exports.args = function (test) {
     )
     iterator.end(t.end.bind(t))
   })
+
+  test('test iterator#next returns this', function (t) {
+    var iterator = db.iterator()
```
Comment (ralphtheninja) [inline abstract/iterator-test.js]: 👍

**Comment by vweevers** [comment]
leveldown is passing.

**Comment by vweevers** [comment]
@ralphtheninja good to release?

**Comment by ralphtheninja** [comment]
Yep. Go!

**Pull request merged**